casper/src/main/scala/coop/rchain/casper/ValidatorIdentity.scala

```diff
@@ -11,7 +11,8 @@ import coop.rchain.shared.{Log, LogSource}
 
 case class ValidatorIdentity(publicKey: Array[Byte],
                              privateKey: Array[Byte],
-                             sigAlgorithm: String) {
+                             sigAlgorithm: String,
+                             shardId: String) {
```

Comment:
I think conceptually the `shardId` does not belong here because read-only nodes (which do not have a validator ID) will still connect to a particular shard. Instead maybe it should be an argument passed to the `Casper` object itself, like the genesis block.